packages/ui-react/src/containers/Layout/Layout.tsx

```diff
@@ -1,224 +1,82 @@
+import HeaderLanguageSwitcher from '../HeaderLanguageSwitcher/HeaderLanguageSwitcher';
+import HeaderUserMenu from '../HeaderUserMenu/HeaderUserMenu';
 
 import styles from './Layout.module.scss';
 
 const Layout = () => {
-  const location = useLocation();
-  const navigate = useNavigate();
-  const { t, i18n } = useTranslation('common');
+  const { t } = useTranslation('common');
 
-  const { config, accessModel, supportedLanguages } = useConfigStore(
-    ({ config, accessModel, supportedLanguages }) => ({ config, accessModel, supportedLanguages }),
+  const { config } = useConfigStore(
+    ({ config, accessModel, supportedLanguages }) => ({
+      config,
+      accessModel,
+      supportedLanguages,
+    }),
     shallow,
   );
-  const userMenuTitleId = useOpaqueId('usermenu-title');
-  const isLoggedIn = !!useAccountStore(({ user }) => user);
-  const favoritesEnabled = !!config.features?.favoritesList;
-  const { menu, assets, siteName, description, features, styling, custom } = config;
-  const metaDescription = description || t('default_description');
+  const { menu, assets, siteName, styling } = config;
   const { footerText: configFooterText } = styling || {};
   const footerText = configFooterText || unicodeToChar(env.APP_FOOTER_TEXT);
 
-  const profileController = getModule(ProfileController, false);
-
-  const { searchPlaylist } = features || {};
-  const hasAppContentSearch = isTruthyCustomParamValue(custom?.appContentSearch);
-  const searchEnabled = !!searchPlaylist || hasAppContentSearch;
-
-  const currentLanguage = useMemo(() => supportedLanguages.find(({ code }) => code === i18n.language), [i18n.language, supportedLanguages]);
-
-  const {
-    query: { data: { collection: profiles = [] } = {} },
-    profilesEnabled,
-  } = useProfiles();
-
-  const selectProfile = useSelectProfile({
-    onSuccess: () => navigate(PATH_HOME),
-    onError: () => navigate(PATH_USER_PROFILES),
-  });
-
-  const { searchQuery, searchActive, userMenuOpen, languageMenuOpen } = useUIStore(
-    ({ searchQuery, searchActive, userMenuOpen, languageMenuOpen }) => ({
-      languageMenuOpen,
-      searchQuery,
-      searchActive,
-      userMenuOpen,
-    }),
-    shallow,
-  );
-  const { updateSearchQuery, resetSearchQuery } = useSearchQueryUpdater();
-  const { profile } = useProfileStore();
-
-  const searchInputRef = useRef<HTMLInputElement>(null) as React.MutableRefObject<HTMLInputElement>;
-
-  const [sideBarOpen, setSideBarOpen] = useState(false);
+  const { sideBarOpen, searchActive } = useUIStore((state) => ({
+    sideBarOpen: state.sideBarOpen,
+    searchActive: state.searchActive,
+  }));
   const banner = assets.banner;
-  const canLogin = accessModel !== ACCESS_MODEL.AVOD;
 
-  useEffect(() => {
-    if (isLoggedIn && profilesEnabled && !profiles?.length) {
-      profileController?.unpersistProfile();
-    }
-    // Trigger once on the initial page load
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
```

Comment:
@AntonLantukh do you remember why this was needed? Can we move this to the controller instead?

Reply:
I don't really remember...It was right in the component, all I did was wrapping it into the `useEffect` hook. I think it may be connected with the case when you have a profile saved in the local storage but no profile on the back-end side. So probably this way we can solve the inconsistency problem.

Reply:
@kiremitrov123 maybe you know why this part was needed?

Reply:
My guess is that this is only needed when switching configs (between integration with and without profiles).